<a href="https://colab.research.google.com/github/Abhishek-009/Innomatics-labs-april_21/blob/main/task_9_hypothesis_testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from scipy.stats import t
import statistics
import matplotlib.pyplot as plt
import numpy as np
from matplotlib import style
style.use('ggplot')

Q-1: Pista House selling Hyderabadi Chicken Dum biryani claims that each parcel packet has 500 grams of biryani (also mentioned on the label of packet). You are sceptic of their claims and believe that on average each packet does not contain 500 grams of biryani. How do you prove your claim?

Step 1:

Alternate Hypothesis :
$$ H_1: \ Average \neq 500 gms $$
Null Hypothesis :
$$ H_0: \ Average = 500 gms $$

 Step 2 :


collect the sample of size N                                                          $$ N = [456,300,489,503,363,497,508,514,313,521]$$

In [ ]:
#computing sample Mean :
population_mean = 500
samples = [456,300,489,503,363,497,508,514,313,521]
sample_mean = (sum(samples)/len(samples))
sample_std = 86.7847912943276
n = 10
sample_mean

446.4

Step 3 : Compute Test Statistics

In [ ]:
statistics.stdev(samples)

86.7847912943276

computed sample mean$$ \bar{x} \ = 446.4\ $$

computed standard deviation$$ {s}  = 86.784\ $$

If population variance is unknown$$ t = \frac{\bar{x} - \mu}{s/\sqrt[2]{n}} $$

Step 4 :

Decide ($\alpha$)  Signidicance level :

In [ ]:
confidence_level = 0.95
alpha = 1 - confidence_level
print(alpha)

0.050000000000000044


Step 5 :
Two-Tailed-Test  $$ H_1: \ Average \neq 500 gms $$

In [ ]:
#caluculation of T_score
import math
def t_score(population_mean,sample_mean,sample_std,n):
    numerator = sample_mean - population_mean
    denominator = sample_std/math.sqrt(n)
    return(numerator/denominator)

In [ ]:
print(t_score(population_mean,sample_mean,sample_std,n))

-1.9530850977123209


Computed T_score$$ {T}  = -1.9530850977123209 $$

In [ ]:
T = -1.9530850977123209

In [ ]:
alpha = 1 - 0.95

t_critical = t.ppf(1-alpha/2, df = 9)

print(t_critical)

2.2621571627409915


$$ Conclusion $$

In [ ]:
if(T < -t_critical):
    print("Reject Null Hypothesis")
else:
    print("Fail to reject Null Hypothesis")

Fail to reject Null Hypothesis


Q-2: You have developed a new Natural Language Processing Algorithms and done a user study. You claim that the average rating given by the users is greater than 4 on a scale of 1 to 5. How do you prove this to your client? 

Step 1:

Alternate Hypothesis :
$$ H_1: \ Average > 4 stars $$
Null Hypothesis :
$$ H_0: \ Average\leq 4 stars $$

Step 2 :

collect the sample of size N                                                          $$ N = [4,3,1,5,5,3,3,3,1,5]$$

In [ ]:
#computing sample Mean :
population_mean_1 = 4
samples_1 = [4,3,1,5,5,3,3,3,1,5]
sample_mean_1 = (sum(samples_1)/len(samples_1))
sample_std_1 = 1.449137674618944
n_1 = 10
sample_mean_1

3.3

Step 3:
Compute Test Statistics

computed sample mean$$ \bar{x_1} \ = 3.3\ $$

In [ ]:
statistics.stdev(samples_1)

1.4944341180973264

computed standard deviation$$ {s_1}  =1.4944341180973264 \ $$

Step 4 :

Decide ($\alpha$)  Significance level :

In [ ]:
confidence_level = 0.95
alpha_1 = 1 - confidence_level
print(alpha_1)

0.050000000000000044


Step 5 : Right-Tailed-Test Because $$ H_1: \ Average > 4 stars $$

In [ ]:
#calculation of T_score
def t_score(population_mean_1,sample_mean_1,sample_std_1,n_1):
    numerator_1 = sample_mean_1 - population_mean_1
    denominator_1 = sample_std_1/math.sqrt(n_1)
    return(numerator_1/denominator_1)
print(t_score(population_mean_1,sample_mean_1,sample_std_1,n_1))

-1.527525231651947


Computed T_score$$ {T_1}  = -1.527525231651947 $$

$$ Right tailed test (H_1) average > 4 rating $$

In [ ]:
T_1 = -1.527525231651947

In [ ]:
alpha_1 = 1 - 0.95

t_critical_1 = t.ppf(1-alpha, df = 9)

print(t_critical_1)

1.8331129326536335


$$ Conclusion $$

In [ ]:
if(T_1 < -t_critical_1):
    print("Reject Null Hypothesis")
else:
    print("Fail to reject Null Hypothesis")

Fail to reject Null Hypothesis


Q-3: TATA has developed a better fuel management system for the SUV segment. They claim that with this system, on average the SUV's mileage is at least 15 km/litre?

Step 1:

Alternate Hypothesis (Bold claim):
$$ H_1:  Average > 15 kmpl $$
Null Hypothesis (Status Quo):
$$ H_0:  Average \leq 15 kmpl $$

Step 2:

collect the sample of size N                                                          $$ N = [14,13,17,15,15,13,18,13,11,15]$$

In [ ]:
#computing sample Mean :
population_mean_2 = 15
samples_2 = [14,13,16,15,12,13,17,13,11,15]
sample_mean_2 = (sum(samples_2)/len(samples_2))
sample_std_2 = 1.8529256146249728
n_2 = 10
sample_mean_2

13.9

Step 3:
Compute Test Statistics

computed sample mean$$ \bar{x_2} \ = 14.4\ $$

In [ ]:
statistics.stdev(samples_2)

1.8529256146249728

computed standard deviation$$ {s_2}  =1.8529256146249728 \ $$

Step 4 :

Decide ($\alpha$)  Significance level :

In [ ]:
confidence_level = 0.90
alpha_2 = 1 - confidence_level
print(alpha_2)

0.09999999999999998


Step 5 : Right-Tailed-Test Because
$$H_1: 𝐴𝑣𝑒𝑟𝑎𝑔𝑒>15 Kmpl$$

In [ ]:
#calculation of T_score
def t_score(population_mean_2,sample_mean_2,sample_std_2,n_2):
    numerator_2 = sample_mean_2 - population_mean_2
    denominator_2 = sample_std_2/math.sqrt(n_2)
    return(numerator_2/denominator_2)
print(t_score(population_mean_2,sample_mean_2,sample_std_2,n_2))

-1.8773044091622946


Computed T_score$$ {T_2}  = -1.8773044091622946 $$

$$ Right tailed test (H_1) average > 15kmpl $$

In [ ]:
T_2 = -1.8773044091622946

In [ ]:
alpha_2 = 1 - 0.95

t_critical_2 = t.ppf(1-alpha, df = 9)

print(t_critical_2)

1.8331129326536335



$$ Conclusion $$


In [ ]:
if(T_2 < -t_critical_2):
    print("Reject Null Hypothesis")
else:
    print("Fail to reject Null Hypothesis")

Reject Null Hypothesis


Q-4: You have developed a new Machine Learning Application and claim that on average it takes less than 100 ms to predict for any future datapoint. How do you convince your client about this claim?

Step1:

Alternate Hypothesis :
$$ H_1: \ Average < 100ms $$
Null Hypothesis :
$$ H_0: \ Average \geq 100ms $$

Step 2:

collect the sample of size N                                                          $$ N = [140,132,99,103,125,113,96,110,96,102]$$

In [ ]:
#computing sample Mean :
population_mean_3 = 100
samples_3 = [140,132,99,103,125,113,96,110,96,102]
sample_mean_3 = (sum(samples_3)/len(samples_3))
sample_std_3 = 111.6
n_3 = 10
sample_mean_3

111.6

Step 3:
Compute Test Statistics

computed sample mean$$ \bar{x_3} \ = 111.6\ $$

In [ ]:
statistics.stdev(samples_3)

15.699964614255947

computed standard deviation$$ {s_3}  =15.699964614255947 \ $$

Step 4 :

Decide ($\alpha$)  Significance level :

In [ ]:
confidence_level = 0.90
alpha_3 = 1 - confidence_level
print(alpha_3)

0.09999999999999998


Step 5 : Left-Tailed-Test Because
$$H_1:𝐴𝑣𝑒𝑟𝑎𝑔𝑒<100𝑚𝑠 $$

In [ ]:
#calculation of T_score
def t_score(population_mean_3,sample_mean_3,sample_std_3,n_3):
    numerator_3 = sample_mean_3 - population_mean_3
    denominator_3 = sample_std_3/math.sqrt(n_3)
    return(numerator_3/denominator_3)
print(t_score(population_mean_3,sample_mean_3,sample_std_3,n_3))

0.3286955274010142


Computed T_score$$ {T_3}  = 0.3286955274010142 $$

In [ ]:
T_3 = 0.3286955274010142

In [ ]:
alpha_3 = 1 - 0.95

t_critical_3 = t.ppf(1-alpha, df = 9)

print(t_critical_3)

1.8331129326536335



$$ Conclusion $$

In [ ]:
if(T_3 < -t_critical_3):
    print("Reject Null Hypothesis")
else:
    print("Fail to reject Null Hypothesis")

Fail to reject Null Hypothesis
